internal/pkg/config/config.go

```diff
@@ -33,12 +33,23 @@ var (
 	// FlagSet should be used by analyzers to reuse -config flag.
 	FlagSet    flag.FlagSet
 	configFile string
+
+	configBytes        []byte
+	configFromBytes    *Config
+	configFromBytesErr error
 )
 
 func init() {
 	FlagSet.StringVar(&configFile, "config", "config.yaml", "path to analysis configuration file")
 }
 
+// SetConfigBytes allows the contents of a configuration file
+// to be provided directly. This is useful when running the tool
+// in an environment where file access is inconvenient.
+func SetConfigBytes(b []byte) {
```

Comment:
Nit: `config.SetConfigBytes` seems a bit against the "no repetitions" naming conventions of Go.  Consider just `config.SetBytes`?  Or expose `configBytes` directly in lieu of an explicit setter.

Reply:
Good point. I like `SetBytes`. Done.